src/app/shared/services/organ-data/organ-data.service.ts

```diff
@@ -29,4 +51,43 @@ export class OrganDataService {
       map(id => id && `${this.pathToImages}/${id}/`)
     );
   }
+
+  /**
+   * Gets all tissue samples
+   * @returns all tissue samples' observable.
+   */
+  getAllTissueSamples(organId: string): Observable<TissueSample[]> {
+    return this.localDatabase.getTissueSamples((item: TissueSample) =>  item.patient.anatomicalLocations[0] === organId);
+  }
+
+  getCounts(organId: string): CountMetadata {
```

Comment:
@KalaRaja I'll merge this, but it should be returning an Observable and using the `async` pipe in the template and is the right way to do this. The .toPromise and returning an object immediately is problematic.

Reply:
yes i will fix this on Friday, also I will replace the filter logic with Filter Builder tool written by Daniel.
The styling is also a little bit off, I will fix those on Friday.